esp-hal/src/dma/buffers.rs

```diff
@@ -12,7 +12,30 @@ pub struct Preparation {
     /// memory?)
     #[cfg_attr(not(esp32s3), allow(dead_code))]
     pub(super) block_size: Option<DmaBufBlkSize>,
-    // burst_mode, alignment, check_owner, etc.
+
+    /// Most DMA channels allow software to configure whether the hardware
+    /// checks that [DmaDescriptor::owner] is set to [Owner::Dma] before
+    /// consuming the descriptor. If this check fails, the channel stops
+    /// operating and fires
+    /// [DmaRxInterrupt::DescriptorError]/[DmaTxInterrupt::DescriptorError].
+    ///
+    /// This field allows buffer implementation to configure this behaviour.
+    ///
+    /// Some buffer implementations may require that the DMA channel performs
+    /// this check before consuming the descriptor to ensure correct
+    /// behaviour. e.g. To prevent wrap-around in a circular transfer.
+    ///
+    /// Some buffer implementations may require that the DMA channel does NOT
+    /// perform this check as the ownership bit will not be set before the
+    /// channel tries to consume the descriptor.
+    ///
+    /// Most implementations won't have any such requirements and will work
+    /// correctly regardless of whether the DMA channel checks or not.
+    ///
+    /// Note: If the DMA channel doesn't support the provided option,
+    /// preparation will fail.
+    pub(super) check_owner: Option<bool>,
```

Comment:
WHy is this optional, what's the difference between None and Some(false)? The comment uses a lot of words to not explain this.

Reply:
- `Some(true)`: DMA channel must the owner bit
- `Some(false)`: DMA channel must NOT the owner bit
- `None`: DMA channel can do whatever it likes/supports.

I can add this after the "This field allows buffer implementation to configure this behaviour." comment.

Reply:
None is somewhat problematic, though, isn't it? It preserves some random previous setting in the current implementation, can't that cause problems?

Reply:
Ehhh, it does preserve some random previous setting, which I admit is a bit weird but I couldn't decide on a default haha. Like the doc says, for some buffer implementations, it doesn't matter whether the DMA channel checks or not.

The main reason this `None` option exists is because SPI DMA (at least, I haven't checked all the other PDMA channels) doesn't support configuring `check_owner`, and I wanted a way for buffers to say "I don't care about this setting".

Reply:
> (And besides, the buffer trait is unsafe so we're trusting the user to pick the right option)

The user can't implement the buffer traits anyway because the fields of Preparation aren't pub 🙃

Reply:
Yeah 😅 I'm working towards that. (Still have to decide on builder pattern vs plain struct)

Reply:
IDK I think "Some(false): DMA channel must NOT [check?] the owner bit" sounds like a reasonable default instead of "None = whatever". You obviously know more here, so I would appreciate you explaining why accidentally owner-checking based on past use is a better option.

Reply:
So there's two parts of the equation:
1. What behavior the buffer implementation needs
2. What the DMA channel can provide

For 1, `None` needs to exist as an option because choosing either true or false means some DMA channels just won't work with the buffer implementation. If the buffer says true, DMA channels that cannot check will fail to prepare the buffer. If the buffer says false, DMA channels that can't not check will fail to prepare the buffer.
Having `None` means the buffer can work with any channel.

For 2, SPI DMA doesn't support configuring the "check owner" bit, so it'll reject false. Every other DMA channel (that is in the hal atm :smile:) supports configuring this. In the `None` case, I left it as "past use" for simplicity but I can set it to true by default I suppose.

To eliminate the "accidentally owner-checking based on past use" I can configure the "check owner" bit to true when the buffer says it doesn't care. Then the behavior would be consistent.